gsl/span

```diff
@@ -347,6 +347,7 @@ class span
 public:
     // constants and types
     using element_type = ElementType;
+    using value_type = ElementType;    
```

Comment:
While you're at it, the `remove_const_t` in the definition of `span_iterator::value_type` should be `remove_cv_t` as well. There's no reason to go out of our way to support `volatile` types in the library, but when it's this easy there's no reason not to do so.

> adding `std::remove_cv_t` causes a lot of compilation errors

Such as?

> `value_type` is implemented in `vector` - it's directly type of the element (cv is not removed)

It's not an issue for containers: they do not support cv-qualified element types (http://stackoverflow.com/questions/6954906/does-c11-allow-vectorconst-t). Were they to do so, I assure you that `vector::value_type` would strip cv-qualifiers from the element type. (Well, it would certainly strip `const`. LWG is very fond of pretending that `volatile` doesn't exist ;))